### ISSUE TITLE
Question for papers?

### ISSUE THREAD

**Issue by SCUTykLin** [tal-ai/MaKE_EMNLP2021#1]
![image](https://user-images.githubusercontent.com/34080744/141433517-fad8d939-fb1d-4a97-a473-02e2aa01fbe7.png)
The red dash line is only enabled in the inference stage. It means it would not be present in the training stage. So How do you calculate KL-divergence in the training stage? Because the zP  is absent because of the absence of red dash line

**Comment by TjoyLiu** [comment]
Hi thanks for your interest, in terms of the KL-divergence in the training stage. It is calculated between the output of the prior net and the output of the posterior net, that is, both are enabled in the training stage. During the inference stage, only red dash line is enabled. Our training code demonstrates the corresponding logic. We need to apologize for the lack of clarity in our paper, thank you for the heads up and we will not be closing this issue.

**Comment by TjoyLiu** [comment]
> ![image](https://user-images.githubusercontent.com/34080744/141433517-fad8d939-fb1d-4a97-a473-02e2aa01fbe7.png) The red dash line is only enabled in the inference stage. It means it would not be present in the training stage. So How do you calculate KL-divergence in the training stage? Because the zP is absent because of the absence of red dash line

Hi thanks for your interest, in terms of the KL-divergence in the training stage. It is calculated between the output of the prior net and the output of the posterior net, that is, both are enabled in the training stage. During the inference stage, only red dash line is enabled. Our training code demonstrates the corresponding logic. We need to apologize for the lack of clarity in our paper, thank you for the heads up and we will not close this issue.

**Comment by SCUTykLin** [comment]
It means in training stage, the Zp and Zq will calculate KL-divergence, after that Zp will abundant and Zq input to decoder. In testing stage, Zp absent and Zq present. right?

**Comment by TjoyLiu** [comment]
![image](https://user-images.githubusercontent.com/47073763/142134377-f52ed649-59cf-424f-be3a-16419817821e.png)
This is the code for training. During training phase, Zq is fed to the decoder for the text generation and the loss calculation. During inference stage, since we do not have gold-standard sentence, we can only input Zp to the decoder, thus Zp present and Zq absent.

**Comment by SCUTykLin** [comment]
Thanks for your reply. I see.
